tests/ast/data/interval.test

```diff
@@ -0,0 +1,81 @@
+## TEST CASE
+
+from snowflake.snowpark._internal.analyzer.expression import Interval
+
+df1 = session.create_dataframe(
+        [
+            [datetime.datetime(2010, 1, 1), datetime.datetime(2011, 1, 1)],
+            [datetime.datetime(2012, 1, 1), datetime.datetime(2013, 1, 1)],
+        ],
+        schema=["a", "b"],
+    )
+
+df2 = df1.select(
+    df1["a"]
+    + Column(
+        Interval(
+            quarter=1,
+            month=1,
+            week=2,
+            day=2,
+            hour=2,
+            minute=3,
+            second=3,
+            millisecond=3,
+            microsecond=4,
+            nanosecond=4,
+        )
+    )
+)
+
+df4 = df1.select(df1["a"] + Column(Interval(1234)))
+
+df5 = df1.select(
+    df1["a"]
+    + Column(
+        Interval(
+            quarter=1,
+            month=2,
+            week=3,
+            day=4,
+            hour=5,
+            minute=6,
+            second=7
+        )
+    )
+)
+
+df6 = df1.select(
+    df1["a"]
+    + Column(
+        Interval(
+            year=1,
+            month=2,
+            week=3,
+            day=4,
+            hour=5,
+            minute=6,
+            second=7,
+            millisecond=8,
+            microsecond=9,
+            nanosecond=10
+        )
+    )
+)
+
+
+## EXPECTED UNPARSER OUTPUT
+
+df1 = session.create_dataframe([[datetime.datetime(2010, 1, 1, 0, 0, 0, 0, tzinfo=datetime.timezone(datetime.timedelta(seconds=-18000), name="EST")), datetime.datetime(2011, 1, 1, 0, 0, 0, 0, tzinfo=datetime.timezone(datetime.timedelta(seconds=-18000), name="EST"))], [datetime.datetime(2012, 1, 1, 0, 0, 0, 0, tzinfo=datetime.timezone(datetime.timedelta(seconds=-18000), name="EST")), datetime.datetime(2013, 1, 1, 0, 0, 0, 0, tzinfo=datetime.timezone(datetime.timedelta(seconds=-18000), name="EST"))]], schema=["a", "b"])
+
+df2 = df1.select(df1["a"] + Interval(quarter=1, month=1, week=2, day=2, hour=2, minute=3, second=3, millisecond=3, microsecond=4, nanosecond=4))
```

Comment:
does this work in snowpark?

Reply:
I believe it does - I pulled this from `test_interval`:
```py
def test_interval(session):
    df1 = session.create_dataframe(
        [
            [datetime.datetime(2010, 1, 1), datetime.datetime(2011, 1, 1)],
            [datetime.datetime(2012, 1, 1), datetime.datetime(2013, 1, 1)],
        ],
        schema=["a", "b"],
    )
    df2 = df1.select(
        df1["a"]
        + make_interval(
            quarters=1,
            months=1,
            weeks=2,
            days=2,
            hours=2,
            minutes=3,
            seconds=3,
            milliseconds=3,
            microseconds=4,
            nanoseconds=4,
        )
    )
    verify_column_result(
        session,
        df2,
        [
            '"(""A"" + INTERVAL \'1 QUARTER,1 MONTH,2 WEEK,2 DAY,2 HOUR,3 MINUTE,3 SECOND,3 MILLISECOND,4 MICROSECOND,4 NANOSECOND\')"',
        ],
        [TimestampType(timezone=TimestampTimeZone.NTZ)],
        None,
    )
```